includes/controllers/pages/class-submit-listing.php

```diff
@@ -136,17 +136,27 @@ public function dispatch() {
         }
 
         if ( ! $this->editing && 'auto-draft' !== get_post_status( $this->listing->get_id() ) ) {
-            $possible_payment = WPBDP_Payment::objects()->filter(
-                array(
-					'listing_id'   => $this->listing->get_id(),
-					'payment_type' => 'initial',
-					'status'       => 'pending',
-                )
-            )->get();
-
-            if ( $possible_payment ) {
-                return $this->_redirect( $possible_payment->get_checkout_url() );
-            }
+			$plan_id = absint( wpbdp_get_var( array( 'param' => 'listing_plan', 'default' => 0 ), 'post' ) );
+			$plan    = wpbdp_get_fee_plan( $plan_id );
+            if ( $plan && $plan->enabled ) {
+				$plan    = $this->get_plan_for_listing();
+				$updated = $this->maybe_update_listing_plan( $plan );
+				if ( $updated ) {
+					// New listing plan changed. We need to delete other payment records.
+					$this->listing->delete_payment_history();
```

Comment:
Let's never delete payment history for successful payments. The database structure for this is pretty rough. I think the problem here is that the plan id should be saved in the payments table.

Reply:
Changed label to `in progress` Need to write some tests on the submission part with the plans